### DISCUSSION TITLE
Create non-closed polyline

### DISCUSSION THREAD

**Discussion by Andreasox** [pymupdf/PyMuPDF#3225] [Looking for help] [answered]
draw_polyline adds automatically a last point equal to the start point thereby creating a closed polyline. 
Is it possible to create a non-closed polyline? A sequence of draw_line is not a feasible solution.

**Comment by JorjMcKie** [comment]
I don't think so:
```python
rect=fitz.Rect(100,100,200,200)
page.draw_polyline((rect.tl,rect.tr,rect.br))
Point(200.0, 200.0)
doc.save("x.pdf")
```
gives this:
![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/b78c1b4e-f995-4e8e-b6f0-ed74b9873c20)

**Reply by Andreasox** [reply]
seems to be that shape.draw_polyline closes the line - I do not understand the difference
linecoords = [(270, 184.0), (270, 212.4), (270, 240.8), (270, 269.2), (300, 283.40000000000003)]
page.draw_polyline(linecoords)
linecoords = [(296.84999999999997, 184.0), (296.84999999999997, 212.4), (296.84999999999997, 240.8), (296.84999999999997, 269.2), (326.84999999999997, 283.40000000000003)]
shape.draw_polyline(linecoords)
shape.finish(width=3, color=color_black)
shape.commit(overlay=False)

<img width="157" alt="image" src="https://github.com/pymupdf/PyMuPDF/assets/6737097/04ab15c9-d41e-4a15-82bb-bb6289b939a3">

**Reply by JorjMcKie** [reply, marked as answer]
Something in between must use "closePath=True".
Default for the finish method is `closePath=True`!

**Reply by Andreasox** [reply]
Of course - thanks
